src/collective/communicator.cu

```diff
@@ -16,7 +17,7 @@ thread_local std::unique_ptr<DeviceCommunicator> Communicator::device_communicat
 
 void Communicator::Finalize() {
   communicator_->Shutdown();
-  communicator_.reset(nullptr);
```

Comment:
Could you please share under which case the communicator can still be called after being shut down? I think a nullptr can be a guard for unintentional calls.

Reply:
I think we have some tests that mix distributed training and local training.